simulator/func_sim/rf/rf.h

```diff
@@ -36,20 +36,39 @@ class RF
     }
 
     template <typename T>
-    void write( Register num, const T& val, bool accumulating_instr = false)
+    void addition( T& new_val, const T& val)
+    {
+      const auto entry_hi = get_entry( Register::mips_hi);
+      const auto entry_lo = get_entry( Register::mips_lo);
+      auto hi_lo = static_cast<uint64>( entry_hi.value);
+      hi_lo <<= 32;
+      hi_lo += static_cast<uint64>( entry_lo.value);
+      hi_lo += static_cast<uint64>( val);
+      new_val = static_cast<T>( hi_lo);
+    }
+
+    template <typename T>
+    void subtraction( T& new_val, const T& val)
+    {
+      const auto entry_hi = get_entry( Register::mips_hi);
+      const auto entry_lo = get_entry( Register::mips_lo);
+      auto hi_lo = static_cast<uint64>( entry_hi.value);
+      hi_lo <<= 32;
+      hi_lo += static_cast<uint64>( entry_lo.value);
+      hi_lo -= static_cast<uint64>( val);
+      new_val = static_cast<T>( hi_lo);
+    }
+
+    template <typename T>
+    void write( Register num, const T& val, int8 accumulating_instr = 0)
     {
         T new_val = val;
         if ( num.is_zero())
             return;
-        if( accumulating_instr){
-            const auto entry_hi = get_entry( Register::mips_hi);
-            const auto entry_lo = get_entry( Register::mips_lo);
-            auto hi_lo = static_cast<uint64>( entry_hi.value);
-            hi_lo <<= 32;
-            hi_lo += static_cast<uint64>( entry_lo.value);
-            hi_lo += static_cast<uint64>( val);
-            new_val = static_cast<T>( hi_lo);
-        }
+        if( accumulating_instr == 1)
+            addition(new_val, val);
+        if( accumulating_instr == -1)
```

Comment:
Add else here

Please follow code guidelines. There should be a space between if and ( as in condition above